tests/src/Functional/AccessTest.php

```diff
@@ -0,0 +1,153 @@
+      'non-member' => ['access content'],
+    ];
+    foreach ($users as $user => $permissions) {
+      $this->users[$user] = $this->drupalCreateUser($permissions, $user);
+    }
+
+    // Create a "group" bundle on the Custom Block entity type and turn it into
+    // a group. Note we're not using the Entity Test entity for this since it
+    // does not have real support for multiple bundles.
+    BlockContentType::create(['type' => 'group']);
+    Og::groupManager()->addGroup('block_content', 'group');
+
+    // Create a group.
+    $this->group = BlockContent::create([
+      'title' => $this->randomString(),
+      'type' => 'group',
+      'uid' => $this->users['owner']->id(),
+    ]);
+    $this->group->save();
+
+    // Create a group content type.
+    $this->createContentType(['type' => 'group_content']);
+    $settings = [
+      'field_storage_config' => [
+        'settings' => [
+          'target_type' => 'block_content',
+        ],
+      ],
+    ];
+    Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, 'node', 'group_content', $settings);
+
+    // Grant members permission to edit their own content.
+    /** @var \Drupal\og\Entity\OgRole $role */
+    $role = $this->container->get('entity_type.manager')
+      ->getStorage('og_role')
+      ->load('block_content-group-member');
+    $role->grantPermission('edit own group_content content');
+    $role->save();
+
+    // Subscribe the member to the group.
+    /** @var \Drupal\og\Entity\OgMembership $membership */
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->users['member']->id())
+      ->setEntityId($this->group->id())
+      ->setGroupEntityType($this->group->getEntityTypeId())
+      ->addRole($role->id())
+      ->save();
+
+    // Create two group content items, one owned by the group owner, and one by
+    // the member.
+    foreach (['owner', 'member'] as $user) {
+      $this->groupContent[$user] = Node::create([
+        'title' => $this->randomString(),
+        'type' => 'group_content',
+        'uid' => $this->users[$user]->id(),
+        OgGroupAudienceHelper::DEFAULT_FIELD => [['target_id' => $this->group->id()]],
+      ]);
+      $this->groupContent[$user]->save();
+    }
+  }
+
+  /**
+   * Tests access to entity operations through the user interface.
+   *
+   * Note that this does not use a dataProvider because of the slow performance
+   * of setting up functional tests.
+   */
+  public function testEntityOperationAccess() {
+    $test_matrix = [
+      // The administrator should have the right to edit both group content
+      // items.
+      'group-admin' => ['owner' => TRUE, 'member' => TRUE],
+      // The member should only have the right to edit their own group content.
+      'member' => ['owner' => FALSE, 'member' => TRUE],
+      // The non-member cannot edit any group content.
+      'non-member' => ['owner' => FALSE, 'member' => FALSE],
```

Comment:
It would be good to add another scenario, we could call:

```php
'wiki-non-member' => ['owner' => FALSE, 'member' => FALSE, 'allow_edit' => TRUE],
```

As the name implies, there are use case that a group would - a la wiki - allow non-members (but not blocked ones) to edit their content. (This is were OG really takes access to the far ends of use cases 😉 )